credsweeper/utils/util.py

```diff
@@ -88,13 +88,82 @@ def get_shannon_entropy(data: str, iterator: str) -> float:
             return 0
 
         entropy = 0.
+        data_len = len(data)
         for x in iterator:
-            p_x = float(data.count(x)) / len(data)
+            p_x = float(data.count(x)) / data_len
             if p_x > 0:
                 entropy += -p_x * math.log(p_x, 2)
 
         return entropy
 
+    """Precalculated data for speedup"""
+    MIN_DATA_ENTROPY: Dict[int, float] = {
+        16: 1.66973671780348,
+        20: 2.07723544540831,
+        32: 3.25392803184602,
+        40: 3.64853567064867,
+        64: 4.57756933688035,
+    }
+
+    @staticmethod
+    def get_min_data_entropy(x: int) -> float:
+        """Returns minimal entropy for size of random data. Precalculated data is applied for speedup"""
+        if x in Util.MIN_DATA_ENTROPY:
+            y = Util.MIN_DATA_ENTROPY[x]
+        elif 8 < x < 64:
+            # approximated for range 12 - 64
+            _x = x - 8
+            y = ((0.000016617804 * _x - 0.002695077) * _x + 0.170393) * _x + 0.4
+        elif 64 < x:
+            # logarithm base 2 - slow, but precise
+            _x = x - 8
+            y = 1.581026279659 * math.log2(_x) - 1.90156
+        else:
+            # less or equal to 8 bytes might have 0 entropy
+            y = 0
+        return y
+
+    @staticmethod
+    def ascii_entropy_small_data_test(data: bytes) -> bool:
```

Comment:
IMO this function name would be good to change with the format which start with `is_`.
As you can see other functions that in `Utils` class have `is_` prefix if it is validating something.
Please check.

Reply:
done ``is_ascii_entropy_validate``

Reply:
Thank you.